reactive/etcd.py

```diff
@@ -428,6 +428,12 @@ def process_snapd_timer():
     # Get the current snapd refresh timer; we know layer-snap has set this
     # when the 'snap.refresh.set' flag is present.
     timer = snap.get(snapname='core', key='refresh.timer').decode('utf-8')
+    if not timer or timer.isspace():
```

Comment:
`isspace` works when the timer gets set to an empty string, which layer-snap eventually returns as `\n`:

```
sudo snap set core refresh.timer=''
...
>>> subprocess.check_output(['sudo', 'snap', 'get', 'core', 'refresh.timer']).decode('utf-8')
'\n'
```

How about we check `timer` against the configured value, as in:
```
if not timer or timer != hookenv.config('snapd_refresh')
```

This would mean that etcd would be in charge of the host snap timer, effectively ignoring any subordinate refresh config, but maybe that's ok?

Reply:
On second thought, I think it's more important to set the refresh timer to whatever it was before the subordinate smashed it.  Blindly falling back to the charm config may not be what we want.

For example, deploy etcd today, Thursday the 18th.  By default, etcd configures a 'max' refresh timer.  This is translated by layer-snap to "wed3" to signify that the snap should not check for refreshes until the 3rd Wednesday (effectively  a month from yesterday).

If we add a subordinate to the etcd unit in a couple weeks, it will nullify the core snap refresh timer, and the leader will fall into this method.  With the current patch in place, etcd will set a new timer based on config (still 'max'), which layer-snap will translate into something other than the previously set 'wed3' timer.

All that to say, let's keep checking timer for None or isspace here. If that condition is true, check for previously set timers with `leader_get` and only fall back to config if we don't have a previous value.

Reply:
Comparing against the config value will catch a situation where the subordinate has a value and the charm does as well. Who should win there? I would vote the main charm.

Reply:
Yeah, I agree that the principal charm should always win.

However, since our config defaults to 'max', it's not as simple as comparing the current timer to config since the current timer will never say 'max'.